src/nvim/screen.c

```diff
@@ -2832,6 +2832,17 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow,
               // when CursorLineNr isn't set?
               char_attr = win_hl_attr(wp, HLF_CLN);
             }
+
+            if (wp->w_p_rnu && lnum < wp->w_cursor.lnum
+                && win_hl_attr(wp, HLF_LNA) > 1) {
```

Comment:
This a bit strange and doesn't fit our model. I would add "default link LineNrAbove LineNr" in `highlight_init_both` in `syntax.c` and get rid of the check.